frontend/src/components/elements/Text/Text.tsx

```diff
@@ -89,30 +98,46 @@ class Text extends React.PureComponent<Props> {
       link: linkWithTargetBlank,
       linkReference: linkReferenceHasParens,
     }
+    const styleProp = { width }
 
     switch (format) {
       // Plain, fixed width text.
-      case TextProto.Format.PLAIN:
-        return (
-          <div className="fixed-width stText" style={{ width }}>
-            {body}
-          </div>
-        )
+      case TextProto.Format.PLAIN: {
+        const props = {
+          className: classNames("fixed-width", "stText"),
+          style: styleProp,
+        }
+
+        return <div {...props}>{body}</div>
+      }
 
       // Markdown.
-      case TextProto.Format.MARKDOWN:
+      case TextProto.Format.MARKDOWN: {
+        const allowHTML = element.get("allowHtml")
+
+        if (allowHTML) {
+          const parseHtml = htmlParser()
+
+          return (
+            <div className="markdown-text-container stText" style={styleProp}>
+              <ReactMarkdown
+                source={body}
+                escapeHtml={false}
+                astPlugins={[parseHtml]}
+              />
+            </div>
+          )
+        }
+
         return (
```

Comment:
Nit: I think it's cleaner / slight more maintainable to have a single return statement:

```
const astPlugins = allowHTML ? [htmlParser()] : null

return (
            <div className="markdown-text-container stText" style={styleProp}>
              <ReactMarkdown
                source={body}
                escapeHtml={false}
                astPlugins={astPlugins}
              />
            </div>
          )
```